libs/jobs/src/lib/jobs.module.ts

```diff
@@ -0,0 +1,34 @@
+import {DynamicModule, Module} from '@nestjs/common'
+import {Scheduler} from './scheduler.service'
+import {PgBossService} from './pg-boss.service'
+import PgBoss from 'pg-boss'
+
+type JobsModuleOptions = {
+  databaseUrl?: string
+}
+
+@Module({
+  controllers: [],
+  providers: [
+    Scheduler,
+    {
+      provide: PgBoss,
+      useExisting: PgBossService
+    }
+  ],
+  exports: []
+})
+export class JobsModule {
+  static forRoot(options?: JobsModuleOptions): DynamicModule {
+    return {
+      module: JobsModule,
+      providers: [
+        {
+          provide: PgBossService,
+          useValue: new PgBossService(options?.databaseUrl ?? process.env['DATABASE_URL']!)
```

Comment:
should probably use `useFactory` here and instead of accessing process.env directly, could use the `ConfigService` https://docs.nestjs.com/techniques/configuration

Reply:
JobsModule would be dependent on it, but not the service. Imo if you access the process.env it should use the ConfigService.